src/hypothesis/strategies.py

```diff
@@ -660,13 +671,37 @@ def characters(whitelist_categories=None, blacklist_categories=None,
                 min_codepoint, max_codepoint
             )
         )
+    if all((whitelist_characters is not None,
+            min_codepoint is None,
+            max_codepoint is None,
+            whitelist_categories is None,
+            blacklist_categories is None,
+            )):
+        raise InvalidArgument(
+            'Cannot have just whitelist_characters=%r alone, '
+            'it would have no effect. Perhaps you want sampled_from()' % (
+                whitelist_characters,
+            )
+        )
+    if (
+        whitelist_characters is not None and
+        blacklist_characters is not None and
+        set(whitelist_characters).intersection(set(whitelist_characters))
```

Comment:
@DRMacIver, I just tracked this down in #776 - if you supply any blacklist and whitelist chars, even if both are empty sets, this fails.  Just a type - switch one instance of whitelist_characters for blacklist_characters and it'll work.

Plus there should probably be a test for this.